### PR TITLE
deps(build): patch version upgrades for k8s dependencies

### PR DESCRIPTION
**Pull request by julio-lopez**
## Change Overview

Upgrades various k8s-related dependencies

## Pull request type

- [x] :hamster: Trivial/Minor

## Test Plan

- [x] :zap: CI

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -101,13 +101,13 @@ require (
...
+	github.com/go-errors/errors v1.4.2 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/go-logr/zapr v1.2.3 // indirect
 	github.com/go-openapi/errors v0.20.2 // indirect
-	github.com/go-openapi/jsonpointer v0.19.5 // indirect
-	github.com/go-openapi/jsonreference v0.20.0 // indirect
-	github.com/go-openapi/swag v0.19.14 // indirect
+	github.com/go-openapi/jsonpointer v0.19.6 // indirect
+	github.com/go-openapi/jsonreference v0.20.1 // indirect
+	github.com/go-openapi/swag v0.22.3 // indirect
```
Comment (julio-lopez) [inline go.mod:110]: This one may cause issues. If it does, we'll see if we can revert this one alone.

**Review by viveksinghggits — APPROVED** [review]

**Comment by pavannd1** [comment]
@julio-lopez Happy to take this forward. Let me know 👍🏼

**Comment by pavannd1** [comment]
I faced similar issues when I was trying out the test changes last week. It passes golint if we revert the `kustomize/kyaml` update.
```
$ make golint
running CMD in the containerized build environment
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Running golangci-lint from config file: .golangci.yml
PASS

PASS
```

**Comment by julio-lopez** [comment]
@pavannd1 Thanks a lot for digging. I just pushed a slightly modified version without the `kyaml` upgrade.

**Pull request merged**